Medipen Description Reagent Display

567Turtle opened the pull request:
## About The Pull Request

Nanotrasen is rolling out their new and improved Medipen Casing 2.0! This all new casing comes with a fancy new LCD screen on the side of the pen which displays the chemicals inside (legally required after the court ruling of Nanotrasen V Hennington) in an attempt to reduce allergic attacks onboard their station due to not properly advertising the contents of their issued medipens.

Thanks to Rohesie#4152 for teaching me a bunch of stuff about DM, couldn't have done this without him.

## Why It's Good For The Game

As stated above everybody hates when they think they're gonna get a free few points from taking a medical allergy, then being on the verge of crit and popping their epi-pen, only to have a vicious allergic reaction do to the 3u of formaldehyde contained inside, who would've known that the epinephrine pen contained something that wasn't epinephrine? Shocking, I know. This update changes that, now before popping your medipen, the advanced scanners inside the pen (recycled pregnancy test parts) will scan the chemicals contained inside the pen and display them on the side of the pen. The hypo spray chassis has not received the same remodel

![image](https://user-images.githubusercontent.com/91508746/184509345-b3bc7e2f-ec40-46e5-bd3a-efed7976d159.png)

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
qol: Changed how medipens descriptions work
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

567Turtle committed:
medipen reagent display

changes how the medipen description works

Kapu1178 commented:
Reduces espionage opportunity. Balance chsnge disguised as qol

567Turtle commented:
> Reduces espionage opportunity. Balance chsnge disguised as qol

You have to be holding the pen to examine it so you can still drive by pen people with 30u of fentanyl, I mean, I guess you can't slide people pens full of fetanyl disguised as epi-pens but... Ehh... If you think it'd be best I could change it to a balance, I don't think it effects gameplay that much though

Pepsilawn commented:
> Reduces espionage opportunity. Balance chsnge disguised as qol

You can't fill medipens with custom reagents, there is no espionage opportunity when the pen's name always reflected what type it was

567Turtle commented:
Didn't even realize you couldn't make custom medipens but yeah, hyposprays, syndie medipens and combat medipens dont receive this update, so you can still inject people instantly with fetanyl using a hypospray or holy water with a combat injector or sleepy juice with a sleepy pen.

Wallemations commented:
I like it but I don't think you need to have the "there is a small lcd screen that reads" part, just having the text reads like a label already.
Hell why is it an lcd screen? That seems like a waste of resources when you could just put a sticker on it or something, y'know

567Turtle commented:
> I like it but I don't think you need to have the "there is a small lcd screen that reads" part, just having the text reads like a label already.

You make a good point yeah but do remember it actively reads out the ingredients inside the pen and turns off when there arent any chemicals because uhh... I thought it was cool... more futuristic y'know? I could change the LCD screen part and remove the blank LCD screen check in the else statement but uhh... I don't really see a good reason to

Mooshimi commented:
> > Reduces espionage opportunity. Balance chsnge disguised as qol
> 
> You can't fill medipens with custom reagents, there is no espionage opportunity when the pen's name always reflected what type it was

you can via the medipen refiller, it requires you to put the regents to refill it with into the machine?
edit: i was wrong and realized this when I looked :) Good to know?
https://github.com/tgstation/tgstation/blob/91db93866e2098489e31dfd068ae01af078ce202/code/game/machinery/medipen_refiller.dm#L1-L14

567Turtle committed:
minor grammar change

nuff said

567Turtle commented:
Just realized I deleted the text that said the pen was spent, gonna undo that really quickly

567Turtle committed:
fixed mistake, custom text for snailpen and pumpup

Fixed a mistake where I forgot to re-add the spent medipen text after removing it for a minute, made it to pumpup gets a custom examine text and the snail pen gets an outdated examine text, those bureaucratic ass-wipes at snail-notrasen need to get with the times instead of using the outdated medipen design.

567Turtle commented:
That should if I did it right be the last commit I do for this PR, if you notice anything wrong with it leave a comment and I'll fix it when I have time, I might've made a mistake while fixing my mistake because I'm sleep deprived to all hell

Wallemations commented:
Why are you getting the length of the list and lowering it instead of just doing a `for(var/[reagent typepath] in [reagent list])`?

567Turtle commented:
> Why are you getting the length of the list and lowering it instead of just doing a `for(var/[reagent typepath] in [reagent list])`?

Simple, I didn't delete or make the list shorter at all that was just for formatting purposes. If I just did that it would've outputted as, "'WARNING: This medipen contains chemical1chemical2chemical3chemical4' in small text."

By make a temporary variable, the i, and then removing the last 2 chemicals from the list of chemicals, second_to_last_reagent and last_reagent, were left with only the first two chemicals. Then I take the first reagent, chemical1 and I add a comma and a space behind it, loop back around to chemical2 and add a comma and a space behind it, and because I removed chemical3 and chemical4 that's where the loop would end, then it moves to where I finish with the last chemicals, just adding a space, putting the name of chemical3, adding a space, adding and, adding a space then putting the name of chemical 4, then adding a period, then putting my reagent_text to equal the now finished reagent_names_list followed by .join("") I get the desired result which you see in the screenshot of "this medipen contains Chemical 1, Chemical 2, Chemical 3 and Chemical 4."

At least I think so, you could be referring to something entirely different and I'm making myself look like a fool.

567Turtle commented:
I uhh... Don't know why all those checks are failing... Odd

567Turtle committed:
changes nothing

literally changes nothing I just want the tests to stop failing :(

Wallemations commented:
> > Why are you getting the length of the list and lowering it instead of just doing a `for(var/[reagent typepath] in [reagent list])`?
> 
> 
> 
> Simple, I didn't delete or make the list shorter at all that was just for formatting purposes. If I just did that it would've outputted as, "'WARNING: This medipen contains chemical1chemical2chemical3chemical4' in small text." or something similar to that.
> 
> 
> 
> By make a temporary variable, the i, and then removing the last 2 chemicals from the list of chemicals, second_to_last_reagent and last_reagent, were left with only the first two chemicals. Then I take the first reagent, chemical1 and I add a comma and a space behind it, loop back around to chemical2 and add a comma and a space behind it, and because I removed chemical3 and chemical4 that's where the loop would end, then it moves to where I finish with the last chemicals, just adding a space, putting the name of chemical3, adding a space, adding and, adding a space then putting the name of chemical 4, then adding a period, then putting my reagent_text to equal the now finished reagent_names_list followed by .join("") I get the desired result which you see in the screenshot of "this medipen contains Chemical 1, Chemical 2, Chemical 3 and Chemical 4."
> 
> 
> 
> At least I think so, you could be referring to something entirely different and I'm making myself look like a fool.

Still can't say I understand how removing some of the reagents from the list helps with this

567Turtle commented:
> > > Why are you getting the length of the list and lowering it instead of just doing a `for(var/[reagent typepath] in [reagent list])`?
> > 
> > 
> > Simple, I didn't delete or make the list shorter at all that was just for formatting purposes. If I just did that it would've outputted as, "'WARNING: This medipen contains chemical1chemical2chemical3chemical4' in small text." or something similar to that.
> > By make a temporary variable, the i, and then removing the last 2 chemicals from the list of chemicals, second_to_last_reagent and last_reagent, were left with only the first two chemicals. Then I take the first reagent, chemical1 and I add a comma and a space behind it, loop back around to chemical2 and add a comma and a space behind it, and because I removed chemical3 and chemical4 that's where the loop would end, then it moves to where I finish with the last chemicals, just adding a space, putting the name of chemical3, adding a space, adding and, adding a space then putting the name of chemical 4, then adding a period, then putting my reagent_text to equal the now finished reagent_names_list followed by .join("") I get the desired result which you see in the screenshot of "this medipen contains Chemical 1, Chemical 2, Chemical 3 and Chemical 4."
> > At least I think so, you could be referring to something entirely different and I'm making myself look like a fool.
> 
> Still can't say I understand how removing some of the reagents from the list helps with this

I didn't remove anything from the list...
Let me explain this in the simplest way I possibly can

Say we have a medipen with 4 chemicals, Hunk, Hoong, Jing and Honk.
`else`
`			var/len = reagents.reagent_list.len`
`			var/list/reagent_names_list = list()`
`			for(var/i in 1 to (len - 2))`
`				var/datum/reagent/reagent_object = reagents.reagent_list[i]`
`				reagent_names_list += "[reagent_object.name], "`

not what its saying in human is:
if theres more then one reagent
      (definition of something unimportant)
      (definition of something unimportant)
      take the list of reagents and remove the last two reagent TEMPORARILY, Jing and Honk
            place Hunk into reagent_text with a comma and a space after it
            place Hoong into reagent_text with a comma and a space after it

Now, for formatting reasons we removed the last two items in the list temporarily for the loop, giving us "Hunk, Hoong, " in simple text

Then for the next part we take the two chemicals we 'deleted', Jing and Honk and place them into the simple text with a different system, instead of adding a comma and a space after each chemical, we just add a space, the word and and another space inbetween the two, leaving us with the desired output for reagent_text, "Hunk, Hoong, Jing and Honk "

ZeWaka commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -138,10 +138,28 @@
+			reagent_text = first.name
+
+		else
+			var/len = reagents.reagent_list.len
```
```suggestion
			var/len = length(reagents.reagent_list)
```
you should also name this better

Turtle567 replied:
now that I'm looking at it I don't think that this line is actually used, I left it in there accidentaly I think

ZeWaka commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -138,10 +138,28 @@
+
 	else
-		. += span_notice("It is spent.")
+		. += span_notice("Theres a blank LCD screen on the side of the medipen.")
```
```suggestion
		. += span_notice("There's a blank LCD screen on the side of the medipen.")
```

ZeWaka commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -138,10 +138,28 @@
+			reagent_names_list += "[second_to_last_reagent.name] and [last_reagent.name]"
+			reagent_text = reagent_names_list.Join("")
+
+		. += span_notice("Theres a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [reagent_text]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```
```suggestion
		. += span_notice("There's a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [reagent_text]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```

567Turtle replied:
Oh I didn't know you could put apostrophes in descriptions, I thought it shmucked up like it does in announcements and such, thanks

ZeWaka replied:
oh you might need to escape it yeah :flooshed:

ZeWaka reviewed (commented):
this whole code helper could be rolled into a helper function like i said in coderbus

567Turtle committed:
oxford comma

In English-language punctuation, a serial comma (also called a series comma, Oxford comma, or Harvard comma)[1][2] is a comma placed immediately after the penultimate term (i.e., before the coordinating conjunction, such as and or or) in a series of three or more terms. For example, a list of three countries might be punctuated either as "France, Italy and Spain" (without the serial comma) or "France, Italy, and Spain" (with the serial comma).[3][4][5]

Opinions among writers and editors differ on whether to use the serial comma, and usage also differs somewhat between regional varieties of English. British English allows constructions with or without this comma,[6] while in American English it is common and sometimes even considered mandatory to use the comma.[7] A majority of American style guides mandate use of the serial comma. These include: APA style,[8] The Chicago Manual of Style, Garner's Modern American Usage,[9] The MLA Style Manual, Strunk and White's The Elements of Style,[10] and the U.S. Government Printing Office Style Manual. By contrast, the Associated Press Stylebook and The New York Times Style Book[11] advise against it. In Canada, the stylebook published by The Canadian Press advises against it. Most British style guides do not mandate its use. The Economist Style Guide notes that most British writers use it only where necessary to avoid ambiguity.[12] A few British style guides mandate it, most notably The Oxford Style Manual (hence the name, "Oxford comma").[13] However, the University of Oxford Style Guide (2014) advises against its use.[14]

The Oxford Companion to the English Language notes: "Usage varies as to the inclusion of a comma before and in the last item. ... This practice is controversial and is known as the serial comma or Oxford comma, because it is part of the house style of Oxford University Press."[15]

567Turtle commented:
Not my fault nobodies been doing anything, I did everything that was asked of me I don't really know why this hasn't been merged yet

ShizCalev commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -138,10 +138,28 @@
 	. = ..()
-	if(reagents?.reagent_list.len)
-		. += span_notice("It is currently loaded.")
+	if(length(reagents?.reagent_list))
+		var/reagent_text = ""
+
+		if(length(reagents.reagent_list) == 1)
+			var/datum/reagent/first = reagents.reagent_list[1]
+			reagent_text = first.name
+
+		else
+			var/len = length(reagents.reagent_list)
+			var/list/reagent_names_list = list()
+			for(var/i in 1 to (len - 2))
+				var/datum/reagent/reagent_object = reagents.reagent_list[i]
+				reagent_names_list += "[reagent_object.name], "
+			var/datum/reagent/last_reagent = reagents.reagent_list[len]
+			var/datum/reagent/second_to_last_reagent = reagents.reagent_list[len - 1]
+			reagent_names_list += "[second_to_last_reagent.name], and [last_reagent.name]"
+			reagent_text = reagent_names_list.Join("")
+
+		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [reagent_text]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```
```suggestion
	if(length(reagents?.reagent_list))
		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```

567Turtle replied:
no fucking way thats all I had to do

567Turtle replied:
im testing the code right now I swear to god

567Turtle replied:
While I am infinitely angered at the fact that this worked... I just want a list of the names of the chemicals... You do not understand my emotion right now it's just pure anger

ShizCalev replied:
```
	if(length(reagents?.reagent_list))
		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list, TRUE, ", ", TRUE)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED

/proc/pretty_string_from_reagent_list(list/reagent_list, names_only, join_text = " | ", final_and)
	//Convert reagent list to a printable string for logging etc
	var/list/rs = list()
	var/list_length = reagent_list.len
	for (var/datum/reagent/R in reagent_list)
		list_length--
		if(final_end && list_length == 1)
			rs += "and [R.name][names_only ? null : ", [R.volume]"]"
		else
			rs += "[R.name][names_only ? null : ", [R.volume]"]"


	return rs.Join(join_text)
```

lets adapt the proc itself to be a bit more robust in that case.

567Turtle replied:
But why? It accomplishes the same thing as the current code, albeit a microscopic optimization, I just don't see a reason to

ShizCalev commented:
> Not my fault nobodies been doing anything, I did everything that was asked of me I don't really know why this hasn't been merged yet

that's an automated bot that just posts if there's no activity in a while.

567Turtle commented:
ShizCalev, while I am infinitely angered that your utter simplification of the code worked perfectly and was so much shorter, I simply don't want to do it because I just want to show the name of the reagent, not how much is in there, that's not needed, plus I think my version was formatted better.

ShizCalev commented:
```
	if(length(reagents?.reagent_list))
		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list, TRUE, ", ", TRUE)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED

/proc/pretty_string_from_reagent_list(list/reagent_list, names_only, join_text = " | ", final_and)
	//Convert reagent list to a printable string for logging etc
	var/list/rs = list()
	var/list_length = reagent_list.len
	for (var/datum/reagent/R in reagent_list)
		list_length--
		if(final_and && list_length == 1)
			rs += "and [R.name][names_only ? null : ", [R.volume]"]"
		else
			rs += "[R.name][names_only ? null : ", [R.volume]"]"


	return rs.Join(join_text)
```

lets adapt the pretty_string proc itself to be a bit more robust in that case.

567Turtle commented:
I see where you're coming from here, but I just don't see much of a reason to go through the effort of injecting that code into my branch because it just accomplishes the same thing as the current code, albeit a miniscule optimization and a little bit easier to use by others, but I'm pretty sure the current code _can_ already be used by others, no? I've gone through the hard part of defining whats what and where, I think it's better to leave it mostly unformatted so if somebody who wants to use the code wants to format it differently they can without scouring back through and making their own proc of your proposition.

ShizCalev commented:
>I just don't see much of a reason to go through the effort of injecting that code into my branch because it just accomplishes the same thing as the current code, albeit a miniscule optimization and a little bit easier to use by others

My job as a maintainer is to literally ensure code is as optimized as possible, and as easily usable by others as possible. 

>but I'm pretty sure the current code _can_ already be used by others, no? I've gone through the hard part of defining whats what and where, I think it's better to leave it mostly unformatted so if somebody who wants to use the code wants to format it differently they can without scouring back through

You're literally making people scour though the code for your specific formatting. They would have to find it, and then copypaste it into whatever they're working on to reuse it.

>and making their own proc of your suggested change.

I'm not asking you to make a new proc on this atom. I'm asking you to change the existing `pretty_string_from_reagent_list` proc located in `code\modules\reagents\chemistry\reagents.dm` with my code (you can literally copypaste what I put, it'll work) so /everyone/ can use your formatting, as well as change the formatting however they need on the fly with that single proc.

567Turtle commented:
After reading over your code again you're totally right man, its well past when I was trying to head to bed and my brains conked out, I'll implement these changes tomorrow, right now Im gonna hit the hay.

JohnFulpWillard commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -138,10 +138,28 @@
-	if(reagents?.reagent_list.len)
-		. += span_notice("It is currently loaded.")
+	if(length(reagents?.reagent_list))
+		var/reagent_text = ""
```
```suggestion
		var/reagent_text
```

JohnFulpWillard commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -277,6 +294,14 @@
 	base_icon_state = "snail"
 	list_reagents = list(/datum/reagent/snail = 10)
 
+/obj/item/reagent_containers/hypospray/medipen/snail/examine()		// no LCD screen for the snails :(
```
This will have the LCD screen because you're calling parent

JohnFulpWillard replied:
why did you mark my review as resolved when it isnt

![image](https://user-images.githubusercontent.com/53777086/187841846-55f01020-d29b-4d2f-ace6-b6063b54683d.png)

JohnFulpWillard commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -297,6 +322,11 @@
+/obj/item/reagent_containers/hypospray/medipen/pumpup/examine()
+	. = ..()
+	if(length(reagents?.reagent_list))
+		. += span_notice("Theres a small LCD screen on the side of the medipen which reads, 'ERROR: Unidentified chemical contained inside medipen, use with caution!' in small text.")
```
Same here

567Turtle replied:
Yeah I think I could but there just doesn't seem to be a reason to, it'd bulk up the code some more all to make it so I don't have to write two lines of code which I already have wrote, both of which have different examine texts then the other, so I'd have to write two different true/false statements, one for the glitching LCD screen and one for the old medipen examine... At least I think, I could be totally wrong

JohnFulpWillard replied:
Make it a var on the parent to avoid showing parent's description

567Turtle replied:
Ok, yeah sure Ill do that in a few hours

567Turtle commented:
if(length(reagents?.reagent_list))
		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list, TRUE, ", ", TRUE, TRUE)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED

/proc/pretty_string_from_reagent_list(list/reagent_list, names_only, join_text = " | ", final_and, capitalize_names)
	//Convert reagent list to a printable string for logging etc
	var/list/rs = list()
	var/reagents_left = reagent_list.len
	var/intial_list_length = reagents_left
	for (var/datum/reagent/R in reagent_list)
		reagents_left--
		if(final_and && intial_list_length > 1 && reagents_left == 0)
			rs += "and [capitalize_names ? capitalize(R.name) : R.name][names_only ? null : ", [R.volume]"]"
		else
			rs += "[capitalize_names ? capitalize(R.name) : R.name][names_only ? null : ", [R.volume]"]"

	return rs.Join(join_text)

Oh and while I'm looking at my github page I might as well add theres at least 10 undefined variables, more like 15 but I'm too lazy to go and count and the end result after being compiled is the same result as your first requested change.

ShizCalev commented:
@567Turtle Tested it locally using your branch and my code DOES work, and DOES output the same as yours. 

![image](https://user-images.githubusercontent.com/6209658/186778047-1d11cf12-05dd-4763-a3ef-23040b39d0ef.png)

Feel free to contact me on discord if you need help implementing it. Or if you want me to just do it, feel free to enable maintainers pushing commits to your branch on the right hand side.

567Turtle committed:
optimizations, foreseeable future, ease of use oh my!

changes how the petty_string from reagents.dm works and optimizes the code a bunch :smiley:

567Turtle committed:
forgot explination

added explination :)

567Turtle commented:
I think this is pretty much a closed and done PR if somebody could take away the draft tag, I'd say its ready to merge

567Turtle commented:
Yipee a check failed which has absolutely nothing to do with anything I changed in the cide, I'll recompile tomrorrow I'm exhausted

ShizCalev commented:
@567Turtle No worries we can force it to recheck.

ShizCalev commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -138,10 +138,10 @@
-	if(reagents?.reagent_list.len)
-		. += span_notice("It is currently loaded.")
+	if(length(reagents?.reagent_list))
+		. += span_notice("There\'s a small LCD screen on the side of the medipen which reads, 'WARNING: This medipen contains [pretty_string_from_reagent_list(reagents.reagent_list, TRUE, ", ", TRUE, TRUE)]. Do not use if allergic to any listed chemicals.' in small text.") //cheapskates at nanotrasen couldnt ball out for LED
```
oh wait actually forgot. so since this is a child it will inherit the examine message from the parent. we'll need to make a new proc to handle this. call it /obj/item/reagent_containers/hypospray/medipen/proc/examine_text_lcd_output or something. put the output of that in this spot, and then change what each subtype says in that.

567Turtle replied:
Tested it and I dont think youre right if you are saying what I think youre saying, the two medipens which I gave a custom examine text have their custom examine texts still and hyposprays dont recieve this examine text

567Turtle replied:
All the children inheriting the examine message from the parent is the intended output I ensure, there's two medipens which I don't want to have this effect and it's easy to override them with 2 lines of code.

567Turtle committed:
ignore this

does absolutely nothing just have to fix shitty broken test thingamajigs

567Turtle committed:
ignore this

does nothing

567Turtle committed:
maybe this fixes it

changed indentation

567Turtle commented:
![image](https://user-images.githubusercontent.com/91508746/187048714-b5fb0c24-e055-4c8e-b512-67b3584e8e7d.png)

567Turtle commented:
im going fucking insane

567Turtle committed:
fixes indentation

i think

optimumtact commented:
The fact you talk about medical allergy as being a "few free points" tells me everything I need to know about you and this change.

567Turtle commented:
be me
make throwaway comment in pr description
pr closed
